requests_pkcs12.py

```diff
@@ -165,20 +174,46 @@ def selftest():
         cryptography.hazmat.primitives.hashes.SHA512(),
         cryptography.hazmat.backends.default_backend()
     )
-    pkcs12_data = cryptography.hazmat.primitives.serialization.pkcs12.serialize_key_and_certificates(
-        name=b'test',
-        key=key,
-        cert=cert,
-        cas=[cert, cert, cert],
-        encryption_algorithm=cryptography.hazmat.primitives.serialization.BestAvailableEncryption(b'correcthorsebatterystaple')
-    )
-    response = get(
-        'https://example.com/',
-        pkcs12_data=pkcs12_data,
-        pkcs12_password='correcthorsebatterystaple'
-    )
-    if response.status_code != 200:
-        raise Exception('Unexpected response: {response!r}'.format(**locals()))
+
+    test_cases = {
+        "withEncryption": {
+            "pkcs12_password": b"correcthorsebatterystaple",
+            "expected_status_code": 200,
+        },
+        "withEmptyPassword": {
+            "pkcs12_password": b"",
+            "expected_status_code": 200,
+        },
+        "withoutEncryption": {
+            "pkcs12_password": None,
+            "expected_status_code": 200,
+        },
+    }
+    for test_case_name, test_case in test_cases.items():
```

Comment:
The code below should be in a separate function, that takes pkcs12_password, expected_status_code and expected_exception_message as parameter.